src/components/section/form-section.component.tsx

```diff
@@ -45,10 +77,29 @@ const FormSection = ({ fields, onFieldChange }) => {
           .map((entry, index) => {
             const { fieldComponent: FieldComponent, fieldDescriptor, handler } = entry;
             const rendering = fieldDescriptor.questionOptions.rendering;
+
+            const historicalValue = fieldDescriptor.historicalExpression
+              ? evaluateExpression(
+                  fieldDescriptor.historicalExpression,
+                  { value: fieldDescriptor, type: 'field' },
+                  fieldsFromEncounter,
+                  encounterContext.initValues,
+                  {
+                    mode: encounterContext.sessionMode,
+                    patient: encounterContext.patient,
+                    previousEncounter: encounterContext.previousEncounter,
+                  },
+                )
+              : null;
+
             const previousFieldValue = encounterContext.previousEncounter
               ? handler?.getPreviousValue(fieldDescriptor, encounterContext.previousEncounter, fieldsFromEncounter)
               : null;
 
+            const transformedHistoricalValue = historicalValue
+              ? historicalValueTransformer(fieldDescriptor, historicalValue)
+              : null;
+
```

Comment:
We shouldn't treat the `historicalValue` vs `previousFieldValue` as separate things, we should instead do something like:

```ts
const previousFieldValue = !isEmpty(fieldDescriptor.historicalExpression)
              ? evaluatePreviousValueFromHD(fieldDescriptor.historicalExpression)
              : pickPreviousValueFromEncounter(encounterContext.previousEncounter);
```